web/main.py

```diff
@@ -74,7 +74,7 @@ def has_permission(email: str, dataset: str, access_file: str | None, bucket_nam
 
 
 @app.route('/<dataset>/<path:filename>')
-def handler(
+def handler(  # noqa: C901
```

Comment:
Adding the extra iterfile function took this over an arbitrary complexity limit, to fix would need to refactor out some of the other code in the handler function, not keen to do that in this PR as it would make the diff and intent of the change unclear.

Reply:
Fine by me.